### PR TITLE
Basic online dynamic FP8 quantization with vLLM

### PR DESCRIPTION
**Pull request by janekl**
# What does this PR do ?

Enabling FP8 quantization for vLLM engines. For now only the basic variant based on on-line dynamic scaling is supported. Depending on a use-case it gives up to 20% inference speedup.

Static quantization will follow in a subsequent MR.

**Collection**: NLP

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Using in nemo_export.py
python tests/export/nemo_export.py \
  --model_name local \
  --model_type llama \
  --checkpoint_dir /models/LLAMA3-8B-base-1.nemo \
  --min_tps 1 \
  --run_accuracy True \
  --test_data_path /data/lambada.json \
  --use_vllm True \
  --vllm_export_kwargs '{"quantization": "fp8"}' \
  --debug

# Using in deploy_vllm_triton.py
python scripts/deploy/nlp/deploy_vllm_triton.py \
  --nemo_checkpoint  /models/LLAMA3-8B-base-1.nemo \
  --model_type llama \
  --tensor_parallelism_size 1 \
  --triton_model_name local \
  --quantization fp8
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to https://github.com/NVIDIA/NeMo/pull/10941

### CODE REVIEW

File: nemo/export/vllm/model_config.py [hunk 1]
```diff
@@ -81,6 +81,7 @@ def __init__(
         self.disable_sliding_window = disable_sliding_window
         self.served_model_name = nemo_checkpoint
         self.multimodal_config = None
+        self.mm_processor_kwargs = {}
```
Comment (janekl) [inline nemo/export/vllm/model_config.py:84]: This is needed for vllm >= 0.6.2, see https://github.com/vllm-project/vllm/blob/v0.6.2/vllm/engine/llm_engine.py#L277

File: nemo/export/vllm/model_config.py [hunk 2]
```diff
@@ -134,6 +135,9 @@ def __init__(
             disable_sliding_window=self.disable_sliding_window,
             sliding_window_len=self.get_hf_config_sliding_window(),
         )
+        self.is_attention_free = self._init_attention_free()
+        self.has_inner_state = self._init_has_inner_state()
```
Comment (janekl) [inline nemo/export/vllm/model_config.py:139]: Needed for vllm >= 0.6.3 https://github.com/vllm-project/vllm/blob/v0.6.3/vllm/config.py#L199-L200

File: nemo/export/vllm/model_loader.py [hunk 3]
```diff
@@ -68,6 +68,9 @@ def _load_nemo_checkpoint_state(nemo_file: str):
 
         return sharded_state_dict
 
+    def download_model(self, model_config: ModelConfig) -> None:
+        raise NotImplementedError
```
Comment (janekl) [inline nemo/export/vllm/model_loader.py:72]: Needed for vllm >= 0.6.1.post1 https://github.com/vllm-project/vllm/blob/v0.6.1.post1/vllm/model_executor/model_loader/loader.py#L190

File: nemo/export/vllm/model_config.py (hunk 2 above)
Comment (oyilmaz-nvidia) [reply nemo/export/vllm/model_config.py:139]: Does this raise error with the vllm = 0.6.0?

Comment (janekl) [reply nemo/export/vllm/model_config.py:139]: No, it's all OK for 0.6.0. It will be needed on bumping vLLM to latest version which I'm planning to do.

**Review by oyilmaz-nvidia — APPROVED** [review]

**Comment by janekl** [comment]
FYI @apanteleev
